lib/ansible/module_utils/urls.py

```diff
@@ -1023,3 +1024,41 @@ def fetch_url(module, url, data=None, headers=None, method=None,
         info.update(dict(msg="An unknown error occurred: %s" % str(e), status=-1))
 
     return r, info
+
+def fetch_file(module, url, data=None, headers=None, method=None,
+              use_proxy=True, force=False, last_mod_time=None, timeout=10, bufsize=65536):
+    '''Download and save a file via HTTP(S) or FTP (needs the module as parameter).
+    This is basically a wrapper around fetch_url().
+
+    :arg module: The AnsibleModule (used to get username, password etc. (s.b.).
+    :arg url:             The url to use.
+
+    :kwarg data:          The data to be sent (in case of POST/PUT).
+    :kwarg headers:       A dict with the request headers.
+    :kwarg method:        "POST", "PUT", etc.
+    :kwarg boolean use_proxy:     Default: True
+    :kwarg boolean force: If True: Do not get a cached copy (Default: False)
+    :kwarg last_mod_time: Default: None
+    :kwarg int timeout:   Default: 10
+    :kwarg int bufsize:   Default: 65536
+
+    :returns: A string, the path to the downloaded file.
+    '''
+    # download file
+    tempdir = tempfile.mkdtemp()
```

Comment:
I talked to Jmainguy on IRC and he thinks that using mkdtemp() should be fine from a permissions standpoint.  There are some other thoughts though:

* Using the module tempdir, the temp dir should be cleaned up automatically by the controller.  If we use a separate tempfile, it becomes the calling code's responsibility to clean up the temp file (and the temp file can be left if the module raises an exception before it cleans up).

* the module doesn't always have a file (in pipelining mode).  When that happens, we can't use __file__ to determine where to place a tempfile.  bcoca proposed adding a mkdtemp/mktemp function that could handle this but that is probably worthy of a separate PR.
*  bcoca also noted that overall, it's better not to have modules do this.  Having two separate tasks works just as well (however, we need to start shipping roles at some point so that composing multiple tasks into a single logical unit is as easy as creating a module).  Perhaps a note in the docstring that modules shouldn't tack on downloading files as a secondary function would be satisfactory for now.